book/models/probabilistic_ode_solver_parameter_estimation.md

```diff
@@ -0,0 +1,364 @@
+
+data_kwargs = {"color": "gray"}
+ax.annotate("Data", (18.25, 40.0), **data_kwargs)
+solution = solve_save_at(theta_true)
+ax = plot_solution(solution, ax=ax, linewidth=4, alpha=0.5, **data_kwargs)
+
+guess_kwargs = {"color": "gray"}
+ax.annotate("Initial guess", (6.0, 12.0), **guess_kwargs)
+solution = solve_save_at(theta_guess)
+ax = plot_solution(solution, ax=ax, linestyle="dashed", alpha=0.75, **guess_kwargs)
+plt.show()
+```
+
+The samples cover a perhaps surprisingly large range of potential initial conditions, but lead to the "correct" data.
+
+In parameter space, this is what it looks like:
+
+```python
+plt.title("Posterior samples (parameter space)")
+plt.plot(states.position[:, 0], states.position[:, 1], "o", alpha=0.5, markersize=4)
+plt.plot(theta_true[0], theta_true[1], "P", label="Truth", markersize=8)
+plt.plot(theta_guess[0], theta_guess[1], "P", label="Initial guess", markersize=8)
+plt.legend()
+plt.show()
+```
+
+Let's add the value of $M$ to the plot to see whether the sampler covers the entire region of interest.
+
+```python
+xlim = 18, jnp.amax(states.position[:, 0]) + 0.5
+ylim = 18, jnp.amax(states.position[:, 1]) + 0.5
+
+xs = jnp.linspace(*xlim, num=300)
+ys = jnp.linspace(*ylim, num=300)
+Xs, Ys = jnp.meshgrid(xs, ys)
+
+Thetas = jnp.stack((Xs, Ys))
+log_M_vmapped_x = jax.vmap(log_M, in_axes=-1, out_axes=-1)
+log_M_vmapped = jax.vmap(log_M_vmapped_x, in_axes=-1, out_axes=-1)
+Zs = log_M_vmapped(Thetas)
+```
+
+```python
+fig, ax = plt.subplots(ncols=2, sharex=True, sharey=True)
+
+ax_samples, ax_heatmap = ax
+
+fig.suptitle("Posterior samples (parameter space)")
+ax_samples.plot(
+    states.position[:, 0], states.position[:, 1], ".", alpha=0.5, markersize=4
+)
+ax_samples.plot(theta_true[0], theta_true[1], "P", label="Truth", markersize=8)
+ax_samples.plot(
+    theta_guess[0], theta_guess[1], "P", label="Initial guess", markersize=8
+)
+ax_samples.legend()
+im = ax_heatmap.contourf(Xs, Ys, jnp.exp(Zs), cmap="cividis", alpha=0.8)
+plt.colorbar(im)
+plt.show()
+```
+
+<!-- #region -->
+Looks great!
+
+## Conclusion
+
+In conclusion, a log-posterior density function can be provided by ProbDiffEq such that any of BlackJAX' samplers yield parameter estimates of IVPs. 
+
+
+## What's next
+
+Try to get a feeling for how the sampler reacts to changing observation noises, solver parameters, and so on.
+We could extend the sampling problem from $\theta \mapsto \log M(\theta)$ to some $(\theta, \sigma) \mapsto \log \tilde M(\theta, \sigma)$, i.e., treat the observation noise as unknown and run Hamiltonian Monte Carlo in a higher-dimensional parameter space.
+We could also add a more suitable prior distribution $p(\theta)$ to regularise the problem.
+
+
+A final side note:
+We could also replace the sampler with an optimisation algorithm and use this procedure to solve boundary value problems (albeit this may not be very efficient; use [this](https://arxiv.org/abs/2106.07761) algorithm instead).
+<!-- #endregion -->
```

Comment:
End the document with 
````
```{bibliography}
:filter: docname in docnames
```
````
to include the bibliography.